imap/search_xapian.c

```diff
@@ -2460,6 +2468,33 @@ static int compare_segs(const void **v1, const void **v2)
     return r;
 }
 
+static int is_indexed_part(xapian_update_receiver_t *tr, const struct message_guid *guid)
+{
+    if (tr->mode == XAPIAN_DBW_XAPINDEXED) {
+        return xapian_dbw_is_indexed(tr->dbw, guid, XAPIAN_WRAP_DOCTYPE_PART);
+    }
+
+    struct conversations_state *cstate = mailbox_get_cstate(tr->super.mailbox);
+    if (!cstate) {
+        xsyslog(LOG_INFO, "can't open conversations for %s",
```

Comment:
%s will come through in the error logs literally here. xsyslog requires a third argument that provides any printf-style arguments.

This also affects the two error messages in activefile_open()

Reply:
Annoyingly, gcc only catches this particular misuse with -Wformat=2 or -Wformat-nonliteral, but we we have cases elsewhere where we legitimately need to use a non-literal format string, so I guess none of us have this warning level enabled usually.

I fixed another case of this that already slipped through to master here: b86e0207d8

Reply:
Fixed and force-pushed a branch update